src/App.js

```diff
@@ -1,25 +1,41 @@
-import { createDeck, shuffle, sliceInThree } from './functions';
+import React, { useState } from 'react';
+import { createDeck, shuffle, sliceDeckInThree, switchStacks, showEleventhCard } from './functions';
+import { CARDS_TO_USE_TRICK_1 } from './constants';
 import Stack from './components/Stack';
 import './styles/global.css';
-const CARDS_TO_SHOW = 21;
+
+const deck = createDeck();
+const shuffledDeck = shuffle(deck);
+const deck21cards = shuffledDeck.slice(0, CARDS_TO_USE_TRICK_1);
+const { stack1, stack2, stack3 } = sliceDeckInThree(deck21cards);
 
 function App() {
-  const deck = createDeck();
-  const shuffledDeck = shuffle(deck);
-  const deck21cards = shuffledDeck.slice(0, CARDS_TO_SHOW);
-  const { initialStack, middleStack, lastStack } = sliceInThree(deck21cards);
+  const [initialStack, setInitialStack] = useState(stack1);
+  const [middleStack, setMiddleStack] = useState(stack2);
+  const [lastStack, setLastStack] = useState(stack3);
+  const [buttonCounter, setButtonCounter] = useState(0);
+
+  const handleClick = (stackName) => {
+    let newDeck = []
+
+    const switchedDeck = switchStacks(newDeck, stackName, initialStack, middleStack, lastStack);
+
+    const { stack1, stack2, stack3 } = sliceDeckInThree(switchedDeck);
 
-  const handleClick = () => {
-    console.log('Cheguei aqui')
-  }
+    setInitialStack(stack1);
+    setMiddleStack(stack2);
+    setLastStack(stack3);
+    setButtonCounter(buttonCounter + 1)
+  };
 
   return (
     <div className='stacks'>  
       <Stack stack={ initialStack } />
       <button
         className='btn'
         type='button'
-        onClick={ handleClick }
+        value='initial'
+        onClick={ ({ target: { value } }) => handleClick(value) }
```

Comment:
É possível simplificar essas linhas, já que você sabe que esse primeiro botão deve sempre retornar `initial`, você pode passar isso direto para a função:

```suggestion
        onClick={ () => handleClick('initial') }
```

Reply:
Feito